e2e/charts/newrelic-infrastructure/templates/daemonset.yaml

```diff
@@ -38,28 +38,63 @@ spec:
 {{ toYaml .Values.image.pullSecrets | indent 8 }}
       {{- end }}
       containers:
-        - name: {{ template "newrelic.name" . }}
-          image: {{ template "newrelic.image" . }}
-          imagePullPolicy: "{{ .Values.image.pullPolicy }}"
+        - name: kubelet-scraper
+          image: {{ .Values.scraperImage.repository }}:{{ .Values.scraperImage.tag }}
+          imagePullPolicy: {{ .Values.scraperImage.pullPolicy }}
           securityContext:
             {{- if .Values.privileged }}
             privileged: true
             {{- else }}
             runAsUser: {{ .Values.runAsUser | default 1000 }}
+            runAsGroup: {{ .Values.runAsGroup | default 2000 }}
             allowPrivilegeEscalation: false
             readOnlyRootFilesystem: true
             {{- end }}
           env:
-            - name: NRIA_LICENSE_KEY
+            - name: "KUBELET_ENABLED"
+              value: "true"
+            - name: "CLUSTER_NAME"
+              value: {{ include "newrelic.cluster" . }}
+            - name: "NRK8S_NODE_NAME"
               valueFrom:
-                secretKeyRef:
-                  {{- if (include "newrelic.licenseKey" .) }}
-                  name: {{ template "newrelic.fullname" . }}-config
-                  key: license
-                  {{- else }}
-                  name: {{ include "newrelic.customSecretName" . }}
-                  key: {{ include "newrelic.customSecretLicenseKey" . }}
-                  {{- end }}
+                fieldRef:
+                  apiVersion: "v1"
+                  fieldPath: "spec.nodeName"
+          volumeMounts:
+            {{- if .Values.privileged }}
+            - name: dev
+              mountPath: /dev
+            - name: log
+              mountPath: /var/log
+            - name: host-volume
+              mountPath: /host
+              readOnly: true
+            {{- else }}
+            - mountPath: /tmp
+              name: tmpfs-tmp
+            - mountPath: /var/cache/nr-kubernetes
+              name: tmpfs-cache
```

Comment:
Could we generate a single volume and use `subPath` instead? Seems cleaner IMO.

Reply:
yes, but likely we are going to garbage collect tmpfs-cache, therefore I would not invest on that now